src/server.ts

```diff
@@ -35,31 +43,40 @@ const listenMessage = `Listening on port: ${config.APP_PORT}. ENV is ${config.EN
  * 5. Get movie by ID
  * */
 app.post("/movies", async (request: Request, response: Response) => {
+  const client = await MongoClient.connect(DB_URL);
+  const db = client.db("moviesDB");
+  const collection = db.collection("movies");
+
   const { name, comment, personalScore } = request.body;
   const data = await OmdbService.getMovie(name);
 
   // movie not found
   if (!data["imdbID"]) {
     // movie not found in IMDB - let's add it to local DB
-    const movie = { imdbID: movieStore.nextId(), Title: name, comment, personalScore };
-    movieStore.add(movie);
+    const movie = {
+      imdbID: movieStore.nextId(),
+      Title: name,
+      comment,
+      personalScore,
+    };
+    collection.insert(movie);
     return response.status(200).json({
       message: `New movie added: ${name}`,
       data: movie,
     });
   }
 
   // movie already exists in DB
-  if (movieStore.exist(data["imdbID"])) {
+  if (collection.findOne({ imdbID: data["imdbID"] })) {
```

Comment:
You can't use async operation `findOne` in if condition. You should make search first and save it result somewhere
```
const movie = await collection.findOne({ imdbID: data["imdbID"] });

if (movie) {....}
```